From 55cb6c604d210489b1b5dc107419fe96307f1534 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 20 May 2016 06:49:45 -0400
Subject: [PATCH] fix smart update failed while proxy used

While using http proxy, it missed to add pycurl.HTTPPROXYTUNNEL:
...
When  an  HTTP proxy is used (-x, --proxy), this option will cause non-HTTP
protocols to attempt to tunnel through the proxy instead of merely using it
to do HTTP-like operations. The tunnel approach is made with the HTTP proxy
CONNECT  request  and requires that the proxy allows direct connect to the
remote port number curl wants to tunnel through to.
...

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 smart/fetcher.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/smart/fetcher.py b/smart/fetcher.py
index dd0a1c9..a24d1b3 100644
--- a/smart/fetcher.py
+++ b/smart/fetcher.py
@@ -1732,6 +1732,8 @@ class PyCurlHandler(FetcherHandler):
 
                         if item.getInfo('proxy'):
                             handle.setopt(pycurl.PROXY, item.getInfo('proxy'))
+                            if item.getInfo('proxy').startswith("http://"):
+                                handle.setopt(pycurl.HTTPPROXYTUNNEL, 1)
                             username=item.getInfo('proxy_username')
                             if username:
                                 handle.setopt(pycurl.PROXYUSERPWD,
-- 
2.8.1

